src/codegen/compile.ml

```diff
@@ -10315,6 +10315,24 @@ and compile_exp_as env ae sr_out e =
   let sr_in, code = compile_exp_with_hint env ae (Some sr_out) e in
   code ^^ StackRep.adjust env sr_in sr_out
 
+and single_case e (cs : Ir.case list) =
```

Comment:
Why not just inline the pattern below and test against the type on the pat, not e?

Reply:
I also learned to mistrust the type note on patterns...